core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -351,10 +364,10 @@ import com.datastax.driver.core.utils.Bytes
             } else {
               val row = rs.one()
               val event = extractEvent(row)
-              if (event.sequenceNr < seqNr) {
+              if (event.sequenceNr < seqNr || pendingFastForward.isDefined && pendingFastForward.get > event.sequenceNr) {
                 // skip event due to fast forward
                 tryPushOne()
-              } else if (config.gapFreeSequenceNumbers && seqNr != event.sequenceNr) {
+              } else if (pendingFastForward.isEmpty && (config.gapFreeSequenceNumbers && seqNr != event.sequenceNr)) {
```

Comment:
a test that combines gaps and fast forward might be good to add, I think this is correct